### PR TITLE
chore: Reorganize `PortfolioBalance`

### PR DESCRIPTION
**Pull request by gambinish**
## **Description**

Step 2 of token sorting is to move `PortfolioBalance` above the `TabView`. This PR introduces this UI change to move the `PortfolioBalance` above the `TabView`, since it can happen separately from the Token sorting task in order to keep PRs small and easy to review / get past CI.

No functional changes. I did bump the font variant per the design spec to `TextVariant.DisplayMD`

## **Related issues**

Design: https://www.figma.com/design/aMYisczaJyEsYl1TYdcPUL/Portfolio-View?node-id=5008-57060&node-type=frame&t=DkOQgh3ZMZfzy6C4-0

## **Screenshots/Recordings**

Screenshot 2024-09-30 at 10 29 54 PM

## **Pre-merge author checklist**

- [x] I’ve followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Mobile Coding Standards](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-mobile/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

### CODE REVIEW

**Comment by gambinish** [comment]
Leaving a comment here to remember to address the following comments left on the origin PR here as a follow up:

https://github.com/MetaMask/metamask-mobile/pull/11471#discussion_r1784009013
https://github.com/MetaMask/metamask-mobile/pull/11471#discussion_r1784015687

File: app/components/UI/Tokens/TokenList/PortfolioBalance/index.test.tsx [hunk 1]
```diff
@@ -0,0 +1,141 @@
+import React from 'react';
```
Comment (vinnyhoward) [inline app/components/UI/Tokens/TokenList/PortfolioBalance/index.test.tsx:1]: Adding new testes for portfolio, love it👍

**Review by vinnyhoward — DISMISSED** [review]
LGTM✅

**Review by salimtb — DISMISSED** [review]

**Comment by salimtb** [comment]
LGTM ✅ great job !

**Review by vinnyhoward — DISMISSED** [review]

**Review by vinnyhoward — DISMISSED** [review]

**Review by Daniel-Cross — DISMISSED** [review]
Looks good!

**Review by salimtb — DISMISSED** [review]

File: app/components/Views/Wallet/__snapshots__/index.test.tsx.snap
Comment (NicolasMassart) [inline app/components/Views/Wallet/__snapshots__/index.test.tsx.snap:1]: Be careful this snapshot is actually rendering the error screen and not the expected feature!

**Review by NicolasMassart — CHANGES_REQUESTED** [review]

File: app/components/Views/Wallet/__snapshots__/index.test.tsx.snap
Comment (gambinish) [reply app/components/Views/Wallet/__snapshots__/index.test.tsx.snap:1]: Thanks for flagging this. I resolved this by only rendering `PortfolioBalance` when the `accountBalanceByChainId` resolves. Lmk if this works for you?

This looks like a very old unit test suite. `should render correctly` seems a bit generic, and also like the outdated unit testing naming convention, which isn't super specific. Should I create an issue to update this suite?

**Review by NicolasMassart — APPROVED** [review]

**Review by salimtb — APPROVED** [review]

**Pull request merged**